src/utils/get-workspaces.ts

```diff
@@ -10,10 +13,25 @@ const WorkspaceModel = zod.optional(
   ])
 );
 
-export function getWorkspaces(
+export async function getWorkspaces(
   packageJSON: Record<string, unknown>
-): Array<string> | null {
-  const result = WorkspaceModel.parse(packageJSON.workspaces);
+): Promise<string[] | null> {
+  let result = WorkspaceModel.parse(packageJSON.workspaces);
+
+  const pnpmWorkspacePath = path.join(process.cwd(), "pnpm-workspace.yaml");
+  const isPnpmWorkspace = await fse.pathExists(pnpmWorkspacePath);
+
+  if (isPnpmWorkspace) {
+    if (result) {
+      throw new Error(
+        "Both `pnpm-workspace.yaml` and `package.json#workspaces` are not supported. Remove `package.json#workspaces` field."
+      );
+    }
+
+    result = jsYaml.load(await fse.readFile(pnpmWorkspacePath, "utf8")) as {
+      packages?: string[];
+    };
+  }
```

Comment:
here magic happens and I think it is better place to validate it in `getWorkspaces` than `validatePackageJson`